modules/sovrnBidAdapter.js

```diff
@@ -72,48 +81,83 @@ export const spec = {
-          ttl: 60
+    try {
+      let sovrnBidResponses = [];
+      if (id &&
+        seatbid &&
+        seatbid.length > 0 &&
+        seatbid[0].bid &&
+        seatbid[0].bid.length > 0) {
+        seatbid[0].bid.map(sovrnBid => {
+          sovrnBidResponses.push({
+            requestId: sovrnBid.impid,
+            cpm: parseFloat(sovrnBid.price),
+            width: parseInt(sovrnBid.w),
+            height: parseInt(sovrnBid.h),
+            creativeId: sovrnBid.crid || sovrnBid.id,
+            dealId: sovrnBid.dealid || null,
+            currency: 'USD',
+            netRevenue: true,
+            mediaType: BANNER,
+            ad: decodeURIComponent(`${sovrnBid.adm}<img src="${sovrnBid.nurl}">`),
+            ttl: 60
+          });
         });
-      });
+      }
+      return sovrnBidResponses
+    } catch (e) {
+      new LogError(e, {id, seatbid}).send()
     }
-    return sovrnBidResponses;
   },
 
   getUserSyncs: function(syncOptions, serverResponses, gdprConsent) {
-    if (serverResponses && serverResponses.length !== 0 && syncOptions.iframeEnabled) {
-      let iidArr = serverResponses.filter(rsp => rsp.body && rsp.body.ext && rsp.body.ext.iid)
-        .map(rsp => { return rsp.body.ext.iid });
-      let consentString = '';
-      if (gdprConsent && gdprConsent.gdprApplies && typeof gdprConsent.consentString === 'string') {
-        consentString = gdprConsent.consentString
-      }
-      if (iidArr[0]) {
-        return [{
-          type: 'iframe',
-          url: '//ap.lijit.com/beacon?informer=' + iidArr[0] + '&gdpr_consent=' + consentString,
-        }];
+    try {
+      if (serverResponses && serverResponses.length !== 0 && syncOptions.iframeEnabled) {
+        let iidArr = serverResponses.filter(rsp => rsp.body && rsp.body.ext && rsp.body.ext.iid)
+          .map(rsp => { return rsp.body.ext.iid });
+        let consentString = '';
+        if (gdprConsent && gdprConsent.gdprApplies && typeof gdprConsent.consentString === 'string') {
+          consentString = gdprConsent.consentString
+        }
+        if (iidArr[0]) {
+          return [{
+            type: 'iframe',
+            url: '//ap.lijit.com/beacon?informer=' + iidArr[0] + '&gdpr_consent=' + consentString,
+          }];
+        }
       }
+      return []
+    } catch (e) {
+      new LogError(e, {syncOptions, serverResponses, gdprConsent}).send()
     }
-    return [];
   },
-};
+}
+
+class LogError {
+  constructor(e, data) {
+    this.error = {}
+    this.error.ts = utils.timestamp()
+    this.error.payload = 'error'
+    this.error.message = e.message
+    this.error.data = data
+    this.error.source = 'sovrnBidAdapter'
+    this.error.stack = e.stack
+    this.error.prebidVersion = $$PREBID_GLOBAL$$.version
+    this.error.url = utils.getTopWindowLocation().href
+    this.error.userAgent = navigator.userAgent
+  }
+  send() {
+    this.error.ts = utils.timestamp()
```

Comment:
Should not be set in the constructor - will fix.

Reply:
i actually think it makes more sense in the constructor than here, so the timestamp more accurately reflects when the error actually occurred, rather than when it happened to get sent